package com.rj.protos.mobile;

message MobileAdResponse {
    required string id = 1; //ID of the bid request to which this is a response.
    required Bidder bidder = 2; // which bidder make the bid
    required string bidId = 3; // the ID generated by the bidder to log or track the bid
    repeated mobileBid bidContent = 4;

message Bidder {
    required string bidderId = 1;
}

message mobileBid {
    required string campaignId = 1;
    optional string biddingType = 2;
    optional string biddingValue = 3;
    optional string currency = 4;
    repeated Creative creative = 5;
    optional Action action = 6;
    optional UUID uuid = 7; //uuid update or anything
	optional string expectCpm = 8; 
}

message Creative {
    optional string creativeId = 1;
    optional string admarkup = 2;
    optional string width = 3;
    optional string height = 4;
    optional string macro = 5; // creative macro to replace in the actions
    repeated TrackingEvents events = 6;
    optional string mediaTypeId = 7;
    optional string mediaSubTypeId = 8;
    optional UUID uuid = 9; //uuid update or anything
}

message Action {
    optional string name = 1;
    optional string inApp = 2;
    optional string content = 3;
    optional string actionType = 4;
}

message TrackingEvents {
    optional string event = 1; // 1. image, telephone, video, 2. start, complet, midpoint,firstquartile, thirdquartile, pause, unpause, mute, unmute, replay, skip, sec
    optional string trackUrl = 2;
}

message UUID {
    repeated UuidType uuidType = 1;
}

enum UuidType {
    FRE = 0;
    SESSION = 1;
}
}
 
